### DISCUSSION TITLE
Errror while Generating the jar file for cassandraTrigger in Knowledge platform

### DISCUSSION THREAD

**Discussion by jashds** [project-sunbird/sunbird-devops#2402] [Q&A]
i am getting error while trying to Generate the jar file for cassandraTrigger as mentioned in 
http://docs.sunbird.org/latest/developer-docs/server-installation/knowledge-platform/index.html

while i followed the introductions provided in this page for server installation 
http://docs.sunbird.org/latest/developer-docs/server-installation/jenkins-setup/   (3.7.0)

Given configuration for 
Global Pipeline Libraries : 
library Name : deploy-conf
Default version : release-2.8.0
Project Repository  :  https://github.com/project-sunbird/sunbird-devops.git

console output for build : 

Started by user user1
Rebuilds build #18
Checking out git https://github.com/project-sunbird/sunbird-learning-platform.git into /var/lib/jenkins/workspace/Build/KnowledgePlatform/CassandraTrigger@script to read pipelines/build/cassandra-trigger/Jenkinsfile
No credentials specified
 > git rev-parse --is-inside-work-tree # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url https://github.com/project-sunbird/sunbird-learning-platform.git # timeout=10
Using shallow fetch with depth 1
Fetching upstream changes from https://github.com/project-sunbird/sunbird-learning-platform.git
 > git --version # timeout=10
 > git fetch --tags --progress --depth=1 -- https://github.com/project-sunbird/sunbird-learning-platform.git +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git rev-parse origin/release-2.8.0^{commit} # timeout=10
Checking out Revision 842ef8eeff8b2838a021c001fdc4c0471bd8183c (origin/release-2.8.0)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 842ef8eeff8b2838a021c001fdc4c0471bd8183c # timeout=10
Commit message: "Merge pull request #1024 from project-sunbird/dock-0.0.1"
 > git rev-list --no-walk 842ef8eeff8b2838a021c001fdc4c0471bd8183c # timeouStarted by user user1
Rebuilds build #18
Checking out git https://github.com/project-sunbird/sunbird-learning-platform.git into /var/lib/jenkins/workspace/Build/KnowledgePlatform/CassandraTrigger@script to read pipelines/build/cassandra-trigger/Jenkinsfile
No credentials specified
 > git rev-parse --is-inside-work-tree # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url https://github.com/project-sunbird/sunbird-learning-platform.git # timeout=10
Using shallow fetch with depth 1
Fetching upstream changes from https://github.com/project-sunbird/sunbird-learning-platform.git
 > git --version # timeout=10
 > git fetch --tags --progress --depth=1 -- https://github.com/project-sunbird/sunbird-learning-platform.git +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git rev-parse origin/release-2.8.0^{commit} # timeout=10
Checking out Revision 842ef8eeff8b2838a021c001fdc4c0471bd8183c (origin/release-2.8.0)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 842ef8eeff8b2838a021c001fdc4c0471bd8183c # timeout=10
Commit message: "Merge pull request #1024 from project-sunbird/dock-0.0.1"
 > git rev-list --no-walk 842ef8eeff8b2838a021c001fdc4c0471bd8183c # timeout=10
Running in Durability level: MAX_SURVIVABILITY
Loading library deploy-conf@release-2.8.0
Attempting to resolve release-2.8.0 from remote references...
 > git --version # timeout=10
 > git ls-remote -h -- https://github.com/project-sunbird/sunbird-devops.git # timeout=10
Could not find release-2.8.0 in remote references. Pulling heads to local for deep search...
 > git rev-parse --is-inside-work-tree # timeout=10
Setting origin to https://github.com/project-sunbird/sunbird-devops.git
 > git config remote.origin.url https://github.com/project-sunbird/sunbird-devops.git # timeout=10
Fetching origin...
Fetching upstream changes from origin
 > git --version # timeout=10
 > git config --get remote.origin.url # timeout=10
 > git fetch --tags --progress -- origin +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git rev-parse release-2.8.0^{commit} # timeout=10
 > git branch -a -v --no-abbrev --contains 25d62e2e10cd5afebd2071b887bf0e17b78f85d4 # timeout=10
Selected match: master revision 25d62e2e10cd5afebd2071b887bf0e17b78f85d4
No credentials specified
 > git rev-parse --is-inside-work-tree # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url https://github.com/project-sunbird/sunbird-devops.git # timeout=10
Fetching without tags
Fetching upstream changes from https://github.com/project-sunbird/sunbird-devops.git
 > git --version # timeout=10
 > git fetch --no-tags --progress -- https://github.com/project-sunbird/sunbird-devops.git +refs/heads/*:refs/remotes/origin/* # timeout=10
Checking out Revision 25d62e2e10cd5afebd2071b887bf0e17b78f85d4 (master)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 25d62e2e10cd5afebd2071b887bf0e17b78f85d4 # timeout=10
Commit message: "Merge pull request #1392 from yravinderkumar33/hawkeye-2.8.6"
 > git rev-list --no-walk 25d62e2e10cd5afebd2071b887bf0e17b78f85d4 # timeout=10
ERROR: Library deploy-conf expected to contain at least one of src or vars directories
org.codehaus.groovy.control.MultipleCompilationErrorsException: startup failed:
WorkflowScript: Loading libraries failed

1 error

	at org.codehaus.groovy.control.ErrorCollector.failIfErrors(ErrorCollector.java:310)
	at org.codehaus.groovy.control.CompilationUnit.applyToPrimaryClassNodes(CompilationUnit.java:1085)
	at org.codehaus.groovy.control.CompilationUnit.doPhaseOperation(CompilationUnit.java:603)
	at org.codehaus.groovy.control.CompilationUnit.processPhaseOperations(CompilationUnit.java:581)
	at org.codehaus.groovy.control.CompilationUnit.compile(CompilationUnit.java:558)
	at groovy.lang.GroovyClassLoader.doParseClass(GroovyClassLoader.java:298)
	at groovy.lang.GroovyClassLoader.parseClass(GroovyClassLoader.java:268)
	at groovy.lang.GroovyShell.parseClass(GroovyShell.java:688)
	at groovy.lang.GroovyShell.parse(GroovyShell.java:700)
	at org.jenkinsci.plugins.workflow.cps.CpsGroovyShell.doParse(CpsGroovyShell.java:142)
	at org.jenkinsci.plugins.workflow.cps.CpsGroovyShell.reparse(CpsGroovyShell.java:127)
	at org.jenkinsci.plugins.workflow.cps.CpsFlowExecution.parseScript(CpsFlowExecution.java:561)
	at org.jenkinsci.plugins.workflow.cps.CpsFlowExecution.start(CpsFlowExecution.java:522)
	at org.jenkinsci.plugins.workflow.job.WorkflowRun.run(WorkflowRun.java:337)
	at hudson.model.ResourceController.execute(ResourceController.java:97)
	at hudson.model.Executor.run(Executor.java:429)
Finished: FAILUREt=10
Running in Durability level: MAX_SURVIVABILITY
Loading library deploy-conf@release-2.8.0
Attempting to resolve release-2.8.0 from remote references...
 > git --version # timeout=10
 > git ls-remote -h -- https://github.com/project-sunbird/sunbird-devops.git # timeout=10
Could not find release-2.8.0 in remote references. Pulling heads to local for deep search...
 > git rev-parse --is-inside-work-tree # timeout=10
Setting origin to https://github.com/project-sunbird/sunbird-devops.git
 > git config remote.origin.url https://github.com/project-sunbird/sunbird-devops.git # timeout=10
Fetching origin...
Fetching upstream changes from origin
 > git --version # timeout=10
 > git config --get remote.origin.url # timeout=10
 > git fetch --tags --progress -- origin +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git rev-parse release-2.8.0^{commit} # timeout=10
 > git branch -a -v --no-abbrev --contains 25d62e2e10cd5afebd2071b887bf0e17b78f85d4 # timeout=10
Selected match: master revision 25d62e2e10cd5afebd2071b887bf0e17b78f85d4
No credentials specified
 > git rev-parse --is-inside-work-tree # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url https://github.com/project-sunbird/sunbird-devops.git # timeout=10
Fetching without tags
Fetching upstream changes from https://github.com/project-sunbird/sunbird-devops.git
 > git --version # timeout=10
 > git fetch --no-tags --progress -- https://github.com/project-sunbird/sunbird-devops.git +refs/heads/*:refs/remotes/origin/* # timeout=10
Checking out Revision 25d62e2e10cd5afebd2071b887bf0e17b78f85d4 (master)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 25d62e2e10cd5afebd2071b887bf0e17b78f85d4 # timeout=10
Commit message: "Merge pull request #1392 from yravinderkumar33/hawkeye-2.8.6"
 > git rev-list --no-walk 25d62e2e10cd5afebd2071b887bf0e17b78f85d4 # timeout=10
ERROR: Library deploy-conf expected to contain at least one of src or vars directories
org.codehaus.groovy.control.MultipleCompilationErrorsException: startup failed:
WorkflowScript: Loading libraries failed

1 error

	at org.codehaus.groovy.control.ErrorCollector.failIfErrors(ErrorCollector.java:310)
	at org.codehaus.groovy.control.CompilationUnit.applyToPrimaryClassNodes(CompilationUnit.java:1085)
	at org.codehaus.groovy.control.CompilationUnit.doPhaseOperation(CompilationUnit.java:603)
	at org.codehaus.groovy.control.CompilationUnit.processPhaseOperations(CompilationUnit.java:581)
	at org.codehaus.groovy.control.CompilationUnit.compile(CompilationUnit.java:558)
	at groovy.lang.GroovyClassLoader.doParseClass(GroovyClassLoader.java:298)
	at groovy.lang.GroovyClassLoader.parseClass(GroovyClassLoader.java:268)
	at groovy.lang.GroovyShell.parseClass(GroovyShell.java:688)
	at groovy.lang.GroovyShell.parse(GroovyShell.java:700)
	at org.jenkinsci.plugins.workflow.cps.CpsGroovyShell.doParse(CpsGroovyShell.java:142)
	at org.jenkinsci.plugins.workflow.cps.CpsGroovyShell.reparse(CpsGroovyShell.java:127)
	at org.jenkinsci.plugins.workflow.cps.CpsFlowExecution.parseScript(CpsFlowExecution.java:561)
	at org.jenkinsci.plugins.workflow.cps.CpsFlowExecution.start(CpsFlowExecution.java:522)
	at org.jenkinsci.plugins.workflow.job.WorkflowRun.run(WorkflowRun.java:337)
	at hudson.model.ResourceController.execute(ResourceController.java:97)
	at hudson.model.Executor.run(Executor.java:429)
Finished: FAILURE

does anyone has any idea about this error, please try to solve

**Comment by joel-ekstep** [comment]
@rjshrjndrn Would you please help with this?
